### DISCUSSION TITLE
Multi node config not working

### DISCUSSION THREAD

**Discussion by thephizzz** [Ride-The-Lightning/RTL#848] [Troubleshooting Questions] [answered]
I have lnd running on a Raspiblitz with RTL and it operates correctly.
RTL is running as a service on the Raspi.
I've added another lnd node running on a W10 PC on the same network.
After editing the RTL-Config.json file to add the new node and stopping and starting the RTL service, the second node information is removed from the RTL-Config.json file and the second node is not recognized.
I assume there is an error in the config file but have tried all options that I know of.
Of note is that my second lnd node does not have a loop.macaroon file so I omitted that line in the Authentication section.
I have attached my multi RTL-Config.json file for review and have added: restlisten=192.xxx.x.xxx:8080 to the second lnd.conf file.
Thank you for any insights.
---------------------------------------------
RTL-Config.json
{
  "port": "3000",
  "defaultNodeIndex": 1,
  "SSO": {
    "rtlSSO": 0,
    "rtlCookiePath": "",
    "logoutRedirectLink": ""
  },
  "nodes": [
    {
      "index": 1,
      "lnNode": "raspivoice",
      "lnImplementation": "LND",
      "Authentication": {
        "macaroonPath": "/home/rtl/.lnd/data/chain/bitcoin/mainnet/",
        "configPath": "/home/rtl/.lnd/lnd.conf",
        "swapMacaroonPath": "/home/rtl/.loop/mainnet/",
        "boltzMacaroonPath": "/home/rtl/.boltz-lnd/macaroons/"
      },
      "Settings": {
        "userPersona": "OPERATOR",
        "themeMode": "DAY",
        "themeColor": "PINK",
        "channelBackupPath": "/home/rtl/RTL-SCB-backup-raspivoice",
        "logLevel": "ERROR",
        "lnServerUrl": "https://localhost:8080",
        "swapServerUrl": "https://localhost:8081",
        "boltzServerUrl": "https://localhost:9003",
        "fiatConversion": false
      }
    },
    {
      "index": 2,
      "lnNode": "lnd XPS",
      "lnImplementation": "LND",
      "Authentication": {
        "macaroonPath": "/mnt/hdd/xps",
        "boltzMacaroonPath": "/mnt/hdd/xps"
      },
      "Settings": {
        "channelBackupPath": "/home/rtl/RTL-SCB-backup-lndXPS",
        "bitcoindConfigPath": "",
        "lnServerUrl": "https://192.168.1.166:8080/v1",
        "boltzServerUrl": "https://192.168.1.166:9003",
        "userPersona": "OPERATOR",
        "themeMode": "DAY",
        "themeColor": "TEAL",
        "fiatConversion": false,
        "logLevel": "ERROR"
      }
    }
  ],
  "multiPassHashed": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
}

**Comment by thephizzz** [comment]
I am using RTL v 0.11.2-beta

**Comment by saubyk** [comment]
Hi @thephizzz are you saying that after you edit the config to include the new node and restart the server, the config setting is getting *removed* from the config file?

If that's what is happening, it is not really a RTL issue and we may need to pull in folks from Raspiblitz. RTL does not make any changes to the config file on its own.

**Reply by thephizzz** [reply]
Hi @saubyk. Yes that is what is happening. I've confirmed several times and made sure that I was working on the correct copy of RTL-Config.json.
Were you able to see any issues with the RTL-Config.json file I uploaded?

**Reply by saubyk** [reply]
@thephizzz your RTL-Config.json file is fine.

@openoms @rootzoll Would you care to weigh-in here please?
Would the user be able to modify RTL config file to add more nodes or are you tightly controlling the configs?

**Comment by thephizzz** [comment, marked as answer]
@Suheb - I did post this on the raspiblitz and received a workaround to comment out a line in the RTL.service.
That did work.
From @openoms: You can comment out or remove the Prestart= line in the rtl.service to preserver your custom changes on restart.

Although on my system the line was: # ExecStartPre=-/home/admin/config.scripts/bonus.rtl.sh prestart lnd mainnet
